RadixWallet/Features/AssetsFeature/Components/StakeUnitList/StakeUnitList.swift

```diff
@@ -0,0 +1,330 @@
+				return .none
+			}
+
+			if state.selectedStakeClaimTokens != nil {
+				guard let stake = state.ownedStakes[id: stakeClaim.validatorAddress],
+				      let ownedStakeClaim = stake.stakeClaimResource
+				else {
+					return .none
+				}
+
+				state.stakedValidators[id: stakeClaim.validatorAddress]?.stakeClaimNFTs?.selectedStakeClaims?.toggle(stakeClaim.token.id)
+				state.selectedStakeClaimTokens?[ownedStakeClaim, default: []].toggle(stakeClaim.token)
+
+				return .none
+			}
+
+			guard let stakeClaimTokens = stakedValidator.stakeDetails.stakeClaimTokens else {
+				return .none
+			}
+
+			state.destination = .stakeClaimDetails(.init(
+				resourceAddress: stakeClaimTokens.resource.resourceAddress,
+				resourceDetails: .success(stakeClaimTokens.resource),
+				token: stakeClaim.token,
+				ledgerState: stakeClaimTokens.resource.atLedgerState,
+				stakeClaim: stakeClaim
+			))
+			return .none
+
+		case let .didTapClaimAll(validatorAddress):
+			guard let stakeClaimTokens = state.stakedValidators[id: validatorAddress]?.stakeDetails.stakeClaimTokens else {
+				return .none
+			}
+
+			return sendStakeClaimTransaction(
+				state.account.address,
+				stakeClaims: [
+					.init(
+						validatorAddress: validatorAddress,
+						resourceAddress: stakeClaimTokens.resource.resourceAddress,
+						ids: stakeClaimTokens.stakeClaims.filter(\.isReadyToBeClaimed).map { $0.id.localId() }
+					),
+				]
+			)
+
+		case .didTapClaimAllStakes:
+			return sendStakeClaimTransaction(
+				state.account.address,
+				stakeClaims: state.stakedValidators.map(\.stakeDetails).compactMap { stake in
+					guard let stakeClaimTokens = stake.stakeClaimTokens,
+					      let stakeClaims = stakeClaimTokens.stakeClaims.filter(\.isReadyToBeClaimed).nilIfEmpty
+					else {
+						return nil
+					}
+
+					return .init(
+						validatorAddress: stake.validator.address,
+						resourceAddress: stakeClaimTokens.resource.resourceAddress,
+						ids: stakeClaims.map { $0.id.localId() }
+					)
+				}
+			)
+		}
+	}
+
+	public func reduce(into state: inout State, internalAction: InternalAction) -> Effect<Action> {
+		switch internalAction {
+		case let .detailsLoaded(.success(details)):
+			state.isLoading = false
+			state.shouldRefresh = false
+			updateAfterLoading(&state, details: details.asIdentifiable())
+			return .none
+		case let .detailsLoaded(.failure(error)):
+			state.isLoading = false
+			state.shouldRefresh = false
+			errorQueue.schedule(error)
+			return .none
+		}
+	}
+
+	public func reduce(into state: inout State, presentedAction: Destination.Action) -> Effect<Action> {
+		switch presentedAction {
+		case let .stakeClaimDetails(.delegate(.tappedClaimStake(stakeClaim))):
+			state.destination = nil
+			return sendStakeClaimTransaction(
+				state.account.address,
+				stakeClaims: [
+					.init(
+						validatorAddress: stakeClaim.validatorAddress,
+						resourceAddress: try! stakeClaim.id.resourceAddress().asSpecific(),
```

Comment:
prefer moving out the `try!` into a computed property, with a rationale why it should be save to use `!` in this context.

Reply:
ups, was really temporary, forgot to change it, thanks for catching it.